exporter/humioexporter/config.go

```diff
@@ -0,0 +1,140 @@
+	// The name of a custom log parser to use, if no parser is associated with the ingest token
+	LogParser string `mapstructure:"log_parser"`
+}
+
+// Humio configuration settings specific to traces
+type TracesConfig struct {
+	// Whether to use Unix timestamps, or to fall back to ISO 8601 formatted strings
+	UnixTimestamps bool `mapstructure:"unix_timestamps"`
+
+	// The time zone to use when representing timestamps in Unix time
+	TimeZone string `mapstructure:"timezone"`
+}
+
+// Humio configuration settings
+type Config struct {
+	// Inherited settings
+	configmodels.ExporterSettings `mapstructure:",squash"`
+	confighttp.HTTPClientSettings `mapstructure:",squash"`
+	exporterhelper.QueueSettings  `mapstructure:"sending_queue"`
+	exporterhelper.RetrySettings  `mapstructure:"retry_on_failure"`
+
+	//Ingest token for identifying and authorizing with a Humio repository
+	IngestToken string `mapstructure:"ingest_token"`
+
+	// Endpoint for the unstructured ingest API, created internally
+	unstructuredEndpoint *url.URL
+
+	// Endpoint for the structured ingest API, created internally
+	structuredEndpoint *url.URL
+
+	// Key-value pairs used to target specific data sources for storage inside Humio
+	Tags map[string]string `mapstructure:"tags,omitempty"`
+
+	// Whether this exporter should automatically add the service name as a tag
+	DisableServiceTag bool `mapstructure:"disable_service_tag"`
+
+	// Configuration options specific to logs
+	Logs LogsConfig `mapstructure:"logs"`
+
+	// Configuration options specific to traces
+	Traces TracesConfig `mapstructure:"traces"`
+}
+
+// Ensure that a valid configuration has been provided, such that we can fail early
+func (c *Config) sanitize() error {
+	if c.IngestToken == "" {
+		return errors.New("requires an ingest_token")
+	}
+
+	if c.Endpoint == "" {
+		return errors.New("requires an endpoint")
+	}
+
+	if c.DisableServiceTag && len(c.Tags) == 0 {
+		return errors.New("requires at least one custom tag when disabling service tag")
+	}
+
+	if c.Traces.UnixTimestamps && c.Traces.TimeZone == "" {
+		return errors.New("requires a time zone when using Unix timestamps")
+	}
+
+	// Ensure that it is possible to construct a URL to access the unstructured ingest API
+	if c.unstructuredEndpoint == nil {
+		endp, err := c.getEndpoint(unstructuredPath)
+		if err != nil {
+			return errors.New("unable to create URL for unstructured ingest API")
+		}
+		c.unstructuredEndpoint = endp
+	}
+
+	// Ensure that it is possible to construct a URL to access the structured ingest API
+	if c.structuredEndpoint == nil {
+		endp, err := c.getEndpoint(structuredPath)
+		if err != nil {
+			return errors.New("unable to create URL for structured ingest API")
+		}
+		c.structuredEndpoint = endp
+	}
+
+	if c.Headers == nil {
+		c.Headers = make(map[string]string)
+	}
+
+	// We require these headers in addition to the ones provided by the user
+	// The user cannot override these headers
+	c.Headers["Content-Type"] = "application/json"
```

Comment:
Did you consider failing to load in case users are setting this?

Reply:
I see. So rather than silently overwriting the users configuration, possibly leaving them confused, I consider certain headers to be an invalid configuration?

Reply:
Yes